core/Pos/Core/Genesis/Canonical.hs

```diff
@@ -63,8 +62,16 @@ instance Buildable SchemaError where
             Just actual -> " but got " <> Builder.fromText actual
         ]
 
-instance (Monad m, Applicative m, MonadError SchemaError m) => ReportSchemaErrors m where
-    expected expec actual = throwError SchemaError
+-- This instance is not reliable. For example, this 'expected' might trigger in 'tryParseString'.
+-- Or might not trigger. I'd prefer to remove this instance. Especially when code perfectly compiles without this instance.
+-- instance (Monad m, Applicative m, MonadError SchemaError m) => ReportSchemaErrors m where
+--     expected expec actual = throwError SchemaError
+--         { seExpected = fromString expec
+--         , seActual = fmap fromString actual
+--         }
+
+instance ReportSchemaErrors (Either SchemaError) where
+    expected expec actual = Left $ SchemaError
```

Comment:
@int-index please review this change.

Reply:
I don't really understand the comment (i.e. why the old instance was unreliable), but the change is OK.

Reply:
I just post here warning I observed with this instance:

```haskell
Pos/Core/Genesis/Canonical.hs:272:8: warning: [-Wsimplifiable-class-constraints]
    • The constraint ‘ReportSchemaErrors m’
        matches an instance declaration
      instance (Monad m, Applicative m, MonadError SchemaError m) =>
               ReportSchemaErrors m
        -- Defined at Pos/Core/Genesis/Canonical.hs:68:10
      This makes type inference for inner bindings fragile;
        either use MonoLocalBinds, or simplify it using the instance
    • In the type signature:
        tryParseString :: forall a m e.
                          (Typeable a, ReportSchemaErrors m, ToString e) =>
                          (Text -> Either e a) -> JSValue -> m a
    |
272 |        forall a m e. (Typeable a, ReportSchemaErrors m, ToString e)
    |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```